src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -186,6 +194,12 @@ private static void ParseData(string resxFilename, bool pathsRelativeToBasePath,
                     case BinSerializedObjectMimeType:
                     case Beta2CompatSerializedObjectMimeType:
                     case CompatBinSerializedObjectMimeType:
+                        // Warn of BinaryFormatter exposure (on by default in .NET 8+)
+                        if (logWarningForBinaryFormatter)
+                        {
+                            log.LogWarningWithCodeFromResources("GenerateResource.BinaryFormatterUse", name, resxFilename, mimetype);
```

Comment:
I think `typename` is of more interest than `mimetype` here. The user likely didn't set the mimetype but let a tool specify that by serializing an object.

On the flip side, it's conceivable that a given object is serializable both via binaryformatter and typeconverter so a different representation of the same object wouldn't throw. Hmm. @merriemcgaw do you have a preference (or a suggestion for someone who might)?

Reply:
I do not personally, but I wonder if @JeremyKuhne and/or @ericstj might have thoughts. My first thought would be to agree `typename` is more interesting but you bring up an interesting case.

Reply:
Since it's for developers why not include both?

Reply:
MimeType here can be one of 3 values, 2 of which haven't been used since like .NET 1.0 days so its probably not all that useful except to help the user know why the resource was being flagged.  Probably could be left out to make a less scary warning and just mentioned in the documentation for the diagnostic ID.  Agreed that typename is much more useful.

Reply:
@ericstj is right -- I forgot those were legacy